cli/admin/src/main/java/io/pravega/cli/admin/segmentstore/FlushToStorageCommand.java

```diff
@@ -46,35 +53,80 @@ public FlushToStorageCommand(CommandArgs args) {
-                new PravegaNodeUri(segmentStoreHost, getServiceConfig().getAdminGatewayPort()), super.authHelper.retrieveMasterToken());
+                new PravegaNodeUri(this.getHosts().get(containerId), getServiceConfig().getAdminGatewayPort()), super.authHelper.retrieveMasterToken());
         reply.get(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
         output("Flushed the Segment Container with containerId %d to Storage.", containerId);
     }
 
     public static CommandDescriptor descriptor() {
         return new CommandDescriptor(COMPONENT, "flush-to-storage", "Persist the given Segment Container into Storage.",
-                new ArgDescriptor("container-id", "The container Id of the Segment Container that needs to be persisted, " +
-                        "if given as \"all\" all the containers will be persisted."),
-                new ArgDescriptor("segmentstore-endpoint", "Address of the Segment Store we want to send this request."));
+                new ArgDescriptor("start-container-id", "The start container Id of the Segment Container that needs to be persisted, " +
+                        "if given as \"all\" all the containers will be persisted."), new ArgDescriptor("end-container-id", "The end container Id of the Segment Container that needs to be persisted, " +
+                "if given as \"all\" all the containers will be persisted."));
+    }
+
+    private Map<Integer, String> getHosts() {
+        Map<Host, Set<Integer>> hostMap;
+        try {
+            @Cleanup
+            ZKHelper zkStoreHelper = ZKHelper.create(getServiceConfig().getZkURL(), getServiceConfig().getClusterName());
+            hostMap = zkStoreHelper.getCurrentHostMap();
+        } catch (Exception e) {
+            System.err.println("Exception accessing to Zookeeper cluster metadata: " + e.getMessage());
+            throw new RuntimeException("Error getting segment store hosts for containers.");
+        }
+
+        Map<Integer, String> containerHostMap = new HashMap<>();
+        for (Map.Entry<Host, Set<Integer>> entry: hostMap.entrySet()) {
+            String ipAddr = entry.getKey().getIpAddr();
+            Set<Integer> containerIds = entry.getValue();
+            for (Integer containerId : containerIds) {
+                containerHostMap.put(containerId, ipAddr);
+            }
+        }
+        return containerHostMap;
+    }
+
+    private void validateArguments() {
+        Preconditions.checkArgument(getArgCount() > 0, "Incorrect argument count.");
+        final String container = getArg(0);
+        if (!NumberUtils.isNumber(container)) {
+            Preconditions.checkArgument(container.equalsIgnoreCase("all"), "Container argument should either be ALL/all or a container id.");
+            Preconditions.checkArgument(getArgCount() == 1, "Incorrect argument count.");
+        } else {
+            final int startContainer = Integer.parseInt(container);
```

Comment:
can use pass negative numbers?

Reply:
User should not pass negative values. Will add validation for that.

Reply:
Added code to validate negative container.

Reply:
Added code to validate negative container.